app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtFilter.java

```diff
@@ -180,13 +185,15 @@ public static List<Group> loadSavedFilters(final Context context) {
                 gp.keys = new ArrayList<>();
                 final JSONArray keysArray = obj.getJSONArray(KEYS);
                 for (int j = 0; j < keysArray.length(); j++) {
-                    gp.keys.add(keysArray.getString(j));
+                    final String key = keysArray.getString(j);
+                    gp.keys.add(NULL_SENTINEL.equals(key) ? null : key);
```

Comment:
We use a null sentinel to store nulls in the json. Without this, 'no priority', 'no context' etc etc filters weren't working